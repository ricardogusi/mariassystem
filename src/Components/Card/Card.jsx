import React from "react";
import styles from "./card.module.css";
import { useState } from "react";
import Modal from '../DataModal/DataModal'
import CardData from '../CardData/CardData'

const Card = () => {

  const [modal, setModal] = useState(true)

  function handleClick() {
    setModal(true)
  }

  return (
    <>
    {modal && <Modal setModal={setModal}/> }
    <div className={styles.card}>
      <button className={styles.adicionarButton} onClick={handleClick} >Adicionar Jogo</button>
      <div className={styles.header}>
        <h3>Jogo</h3>
        <h3>Placar</h3>
        <h3>Mínimo da Temporada</h3>
        <h3>Máximo da Temporada</h3>
        <h3>Quebra recorde Mínimo</h3>
        <h3>Quebra recorde Máximo</h3>
      </div>
      <CardData/>
    </div>
    </>
  );
};

export default Card;
