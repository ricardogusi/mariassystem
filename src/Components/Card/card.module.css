.card {
  background-color: #ffcb94;
  border-radius: 5px;
  width: 190vh;
  height: 55vh;
  display: flex;
  font-family: "Raleway", sans-serif;
  padding: .5rem;
  flex-direction: column;
  position: relative;
  overflow: auto;
}


::-webkit-scrollbar-track {
  background-color: #297894;
}
::-webkit-scrollbar {
  width: 10px;
  
}
::-webkit-scrollbar-thumb {
  background: #E0A86C;
  border-radius: 5px;
}


.header {
  display: grid;  
  grid-template-columns: repeat(6, 1fr);
  text-align: center;    
  width: 190vh;  
  color: #e3af2b;
  font-family: "Raleway", sans-serif;  
  
}


.adicionarButton {
  
  margin-top: 15px;
  border: none;
  border-radius: 5px;
  height: 30px;
  background-color: #fb962a;
  color: white;
  cursor: pointer;
  font-family: "Raleway", sans-serif;
  font-weight: bold;
}

.adicionarButton:hover {
  background-color: #ffcb94;
  color: #297894;
  font-weight: bold;
}



