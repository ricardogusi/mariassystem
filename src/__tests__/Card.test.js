
import React from 'react';
import * as ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Card from '../Components/Card/Card.jsx'



it('Renderiza corretamente', () => {
    const tree = renderer
      .create(<Card /> )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

it('Verifica se contém o título correto', ()=>{
    const root = document.createElement('div');
    ReactDom.render(<Card />, root);
    expect(root.querySelector('h1').textContent).toBe('Just do it!')

})
