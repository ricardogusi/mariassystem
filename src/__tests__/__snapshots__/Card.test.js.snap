// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza corretamente 1`] = `
Array [
  <h1>
    Just do it!
  </h1>,
  <button
    className="adicionarButton"
  >
    Adicionar Jogo
  </button>,
  <div
    className="header"
  >
    <h3>
      Jogo
    </h3>
    <h3>
      Placar
    </h3>
    <h3>
      Mín da Temporada
    </h3>
    <h3>
      Máx da Temporada
    </h3>
    <h3>
      Quebra recorde Mín
    </h3>
    <h3>
      Quebra recorde Máx
    </h3>
  </div>,
  <div
    className="card"
  />,
]
`;
